src/kbmod/filters/stamp_filters.py

```diff
@@ -66,3 +73,174 @@ def keep_row(self, row: ResultRow):
+
+        Returns
+        -------
+        str
+            The filter name.
+        """
+        return (
+            f"StampMomentsFilter_m01_{self.m01_thresh}_m10_{self.m10_thresh}"
+            f"_m11_{self.m11_thresh}_m02_{self.m02_thresh}_m20_{self.m20_thresh}"
+        )
+
+    def keep_row(self, row: ResultRow):
+        """Determine whether to keep an individual row based on
+        the offset of the stamp's peak.
+
+        Parameters
+        ----------
+        row : ResultRow
+            The row to evaluate.
+
+        Returns
+        -------
+        bool
+           An indicator of whether to keep the row.
+        """
+        # Filter any row without a stamp.
+        if row.stamp is None:
+            return False
+
+        # Check the stamp's width is correct.
+        width = 2 * self.stamp_radius + 1
+        if len(row.stamp) != width * width:
+            raise ValueError("Expected stamp of size {width} by {width}.")
+
+        # Find the peack in the image.
+        stamp = row.stamp.reshape([width, width])
+        moments = raw_image(stamp).find_central_moments()
+        return (
+            (abs(moments.m01) < self.m01_thresh)
+            and (abs(moments.m10) < self.m10_thresh)
+            and (abs(moments.m11) < self.m11_thresh)
+            and (moments.m20 < self.m20_thresh)
+            and (moments.m02 < self.m02_thresh)
+        )
+
+
+class StampCenterFilter(Filter):
+    """A filter on whether the center of the stamp is a local
+    maxima and the percentage of the stamp's total flux in this
+    pixel.
+    """
+
+    def __init__(self, stamp_radius, local_max, flux_thresh, *args, **kwargs):
+        """Create a StampCenterFilter.
+
+        Parameters
+        ----------
+        stamp_radius : ``int``
+            The radius of a stamp.
+        local_max : ``bool``
+            Require the central pixel to be a local maximum.
+        flux_thresh : ``float``
+            The fraction of the stamp's total flux that needs to be in
+            the center pixel [0.0, 1.0].
+        """
+        super().__init__(*args, **kwargs)
+
+        if stamp_radius <= 0:
+            raise ValueError("Invalid stamp radius {stamp_radius}.")
+        self.stamp_radius = stamp_radius
+
+        self.local_max = local_max
+        self.flux_thresh = flux_thresh
+
+    def get_filter_name(self):
+        """Get the name of the filter.
+
+        Returns
+        -------
+        str
+            The filter name.
+        """
+        return f"StampCenterFilter_{self.local_max}_{self.flux_thresh}"
+
+    def keep_row(self, row: ResultRow):
+        """Determine whether the center pixel meets the filtering criteria.
+
+        Parameters
+        ----------
+        row : ResultRow
+            The row to evaluate.
+
+        Returns
+        -------
+        bool
+           An indicator of whether to keep the row.
+        """
+        # Filter any row without a stamp.
+        if row.stamp is None:
```

Comment:
There's some duplicated code between these methods, namely, checking that there's a stamp and is the correct size. Would it make sense to either create some `BaseStampFilter` that has this stamp checking, or at least break out into a common method?

Reply:
That's fair. Created an abstract base class and moved the validation steps there.